roc/context.go

```diff
@@ -41,6 +42,11 @@ type Context struct {
 // Allocates and initializes a new context. May start some background threads.
 // User is responsible to call Context.Close to free context resources.
 func OpenContext(config ContextConfig) (*Context, error) {
+	logWrite(LogDebug, fmt.Sprintf("%v", logFormat{op: "roc_context_open_begin()",
+		params: map[string]interface{}{
+			"config":  config,
+			"version": Version(),
+		}}))
```

Comment:
(3) `roc_context_open_begin` looks like an function name, but there is no such function, and it may be confusing. I suggest just to use current Go function everywhere to keep it simple:

```go
logWrite(LogDebug, "entering OpenContext(): config=%+v", config)
```

Reply:
(4) `OpenContext` has multiple return paths, so just adding `logWrite` to the end is incomplete. Instead, we need to use defer:

```go
logWrite(LogDebug, "entering OpenContext(): config=%+v", config)
defer logWrite(LogDebug, "leaving OpenContext()")
```

Reply:
(5) It would be very convenient to log returned error, if it's present. We could use named returns:

```go
func OpenContext(config ContextConfig) (ctx *Context, err error) {
    logWrite(LogDebug, "entering OpenContext(): config=%+v", config)
    defer logWrite(LogDebug, "leaving OpenContext(): err=%v", err)
    ...
```

Reply:
(6) Finally, it would be convenient to log context / sender / receiver pointer when they're created, closed, and in all methods. This will allow distinguishing multiple instances of same type in logs.

```go
func OpenContext(config ContextConfig) (ctx *Context, err error) {
    logWrite(LogDebug, "entering OpenContext(): config=%+v", config)
    defer logWrite(LogDebug, "leaving OpenContext(): context=%p err=%v", ctx, err)
    ...
```

```go
func OpenSender(context *Context, config SenderConfig) (sender *Sender, err error) {
    logWrite(LogDebug, "entering OpenSender(): context=%p config=%+v", context, config)
    defer logWrite(LogDebug, "leaving OpenSender(): context=%p sender=%p err=%v", ctx, sender, err)
    ...
```